modules/Database/Arangojs.js

```diff
@@ -321,6 +322,32 @@ class ArangoJS {
         return 'ArangoJS';
     }
 
+    /**
+    * Get ArangoDB version
+    * @param {string} - host
+    * @param {string} - port
+    * @param {string} - username
+    * @param {string} - password
+    * @returns {Promise<any>}
+    */
+    version(host, port, username, password) {
```

Comment:
We kinda silently switched to the async/awaits. This method should look fairly simpler in async version:
```
async version(host, port, username, password) {        
    const result = await request
            .get(`http://${host}:${port}/_api/version`)
            .auth(username, password);

    if (result.status === 200) {
        return res.body.version;
    } else {
        throw Error('Failed to contact arangodb');
    }
}
```
Same applies for all other methods.

Reply:
will improve it for async/await, thanks @kipliklotrika